src/main/java/org/jboss/remoting3/RemotingXmlParser.java

```diff
@@ -72,18 +79,56 @@ private static void parseDocument(final ConfigurationXMLStreamReader reader, fin
 
     private static void parseEndpointElement(final ConfigurationXMLStreamReader reader, final EndpointBuilder builder) throws ConfigXMLParseException {
         final int attributeCount = reader.getAttributeCount();
+        int readTimeout = -1;
+        int writeTimeout = -1;
+        boolean tcpKeepAlive = false;
+        boolean setTcpKeepAlive = false;
+        int heartbeatInterval = -1;
         for (int i = 0; i < attributeCount; i++) {
             checkAttributeNamespace(reader, i);
             switch (reader.getAttributeLocalName(i)) {
                 case "name": {
                     builder.setEndpointName(reader.getAttributeValueResolved(i));
                     break;
                 }
+                //attributes added to jboss-remoting 5.1, heartbeat, read/write timeout and keepalive
+                case "read-timeout": {
+                    readTimeout = reader.getIntAttributeValueResolved(i, 0, Integer.MAX_VALUE);
+                    break;
+                }
+                case "write-timeout": {
+                    writeTimeout = reader.getIntAttributeValueResolved(i, 0, Integer.MAX_VALUE);
+                    break;
+                }
+                case "tcp-keepalive": {
+                    setTcpKeepAlive = true;
+                    tcpKeepAlive = reader.getBooleanAttributeValueResolved(i);
+                    break;
+                }
+                case "heartbeat-interval": {
+                    heartbeatInterval = reader.getIntAttributeValueResolved(i, 0, Integer.MAX_VALUE);
+                    break;
+                }
                 default: {
                     throw reader.unexpectedAttribute(i);
                 }
             }
         }
+        OptionMap.Builder optionMapBuilder = OptionMap.builder();
+        if (readTimeout != -1L) {
+            optionMapBuilder.set(Options.READ_TIMEOUT, readTimeout);
+        }
+        if (writeTimeout != -1L) {
+            optionMapBuilder.set(Options.WRITE_TIMEOUT, writeTimeout);
+        }
+        if (setTcpKeepAlive) {
+            optionMapBuilder.set(Options.KEEP_ALIVE, tcpKeepAlive);
+        }
+        if (heartbeatInterval != -1) {
+            optionMapBuilder.set(RemotingOptions.HEARTBEAT_INTERVAL, heartbeatInterval);
+        }
+
+        builder.buildDefaultConnectionsOptionMap(optionMapBuilder.getMap());
```

Comment:
builder.setDefaultConnectionsOptionMap would be a more appropriate name here, even with the slight difference in behavior at this method when compared to the simpler setEndpointName, for example